modules/server/src/main/scala/scaladex/server/route/Badges.scala

```diff
@@ -90,17 +90,22 @@ class Badges(database: WebDatabase)(implicit executionContext: ExecutionContext)
   ): RequestContext => Future[RouteResult] =
     parameter("target".?) { binaryVersion =>
       shieldsOptionalSubject { (color, style, logo, logoWidth, subject) =>
-        val res = getSelectedArtifact(
-          organization,
-          repository,
-          binaryVersion,
-          artifactName
-        )
+        val res = binaryVersion match {
+          case Some(binaryVersion) =>
+            getSelectedArtifact(
+              organization,
+              repository,
+              Some(binaryVersion),
+              artifactName
+            ).map(_.map(_.version))
+          case None =>
+            database.getLastVersion(Project.Reference(organization, repository), artifactName).map(Some(_))
```

Comment:
This seems consistent with what the project page does.
https://github.com/scalacenter/scaladex/blob/ebfab5a7a1b760e7eae6cbb55661c3d330144606/modules/server/src/main/scala/scaladex/server/route/ProjectPages.scala#L272-L279